src/NBitcoin/FeeRate.cs

```diff
@@ -41,7 +41,7 @@ public FeeRate(Money feePaid, int size)
             if(feePaid.Satoshi < 0)
                 throw new ArgumentOutOfRangeException("feePaid");
             if(size > 0)
-                this._FeePerK = feePaid / size * 1000;
+                this._FeePerK = (long)(feePaid.Satoshi / (float)size * 1000);
```

Comment:
Maybe

```
this._FeePerK = new Money(feePaid.Satoshi * (1000.0m / size), MoneyUnit.Satoshi);
```

to avoid using float

Reply:
I had to change to decimal because float was wrong (still rounding issue)
btw it uses Money implicit operator but I can use Money constructor too if needed

Reply:
btw yeah I prefer Money constructor that accepts decimal, good point

Reply:
on the other side, money decimal constructor does conversions, while actually I don't need the conversion